### ISSUE TITLE
Issue with zmq grovel (Windows + SBCL)

### ISSUE THREAD

**Issue by jhamell** [galdor/lisp-zmq#17]
Hello,

I initially had an issue when loading lisp-zmq.  I have a particular need to use your software on Windows, and am working within SBCL and Clozure - it has been some time since I worked within Lisp, so this may be a fairly basic mistake on my part.

Within `zmq.lisp`, there are definitions for `:fd`:
```
     (define-sockopt-type :fd #+win32 win32-socket
                              #-win32 :int)
```
When I load the system using `asdf` in SBCL:
```
debugger invoked on a UNBOUND-VARIABLE in thread
#<THREAD "main thread" RUNNING {10030E7723}>:
  The variable WIN32-SOCKET is unbound.
```
If I change the line in `zmq.lisp` to
```
(define-sockopt-type :fd :int)
```
It compiles.  I don't need to make this change with Clozure, but they don't define the same `*features*` there (SBCL sets `win32`, while Clozure uses `windows`, etc).

Thanks,
Josh

**Comment by galdor** [comment]
Apparently I sometimes use `#+windows` (for example in `grovel.lisp`) and some other times `#+win32`. If you provide a patch to fix it for both SBCL and Clozure, I will merge it :)

**Comment by jhamell** [comment]
Hi Nicolas,

Thank you for the response.

I believe the use of `win32-socket` in the definitions unnecessary.  I looked at the source of both sbcl tag 1.2.7 and `master`, and the only string match is for a file name in `contrib/sb-bsd-sockets` which defines the win32 socket functions.  Internally, it treats the socket as an `:int`.

On the offchance that the `sb-bsd-sockets` asdf package was defining the variable, I forced it to load before `zmq` and reverted my local fix.  As expected, it still failed to load giving the `UNBOUND-VARIABLE ... WIN32-SOCKET` error.

In summary, I changed `zmq.lisp` in both places where it defined
```
     (define-sockopt-type :fd #+win32 win32-socket
                              #-win32 :int)
```
with 
```
     (define-sockopt-type :fd :int)
```

Thanks,
Josh
